src/Microsoft.VisualStudio.Composition.Analyzers/VSMEF001PropertyMustHaveSetter.cs

```diff
@@ -40,25 +40,25 @@ public class VSMEF001PropertyMustHaveSetter : DiagnosticAnalyzer
         public override void Initialize(AnalysisContext context)
         {
             context.EnableConcurrentExecution();
-            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```

Comment:
If the intent is to report diagnostics in generated code, then it should be `Analyze | ReportDiagnostics`

Reply:
This strikes me as a bug fix then, is that correct? Is the idea that without this extra flag, we won't report diagnostics on generated code, despite the existing flag we set? It seems odd that `ReportDiagnostics` would be broken if used by itself. Is there documentation or other bugs that I can research to learn more about  this?

Reply:
@AArnott To my understanding, the presence of `Analyze` determines whether or not you get callbacks in generated code, and the presence of `ReportDiagnostics` determines whether or not a reported diagnostic in generated code will be suppressed.

If you want to treat generated code the same as non-generated code (which appears to be the original intent here), you specify both.

There are cases where it's useful to only use `Analyze`, for example, consider an analyzer that reports unused methods, and you don't want to report diagnostics in generated code.

- `Analyze | ReportDiagnostic`: will not work because it reports diagnostics in generated code.
- `None`: will not work, a non-generated method may only be called in generated code. You'll report it as unused while it's used by generated code.
- `Analyze` gives the correct behavior. You don't report unused methods in generated code, but you get callbacks for generated code so that you know whether a non-generated method is used by generated code.

I don't know if `ReportDiagnostics` alone is ever useful.

Reply:
I cannot.  But @mavasani definitely can :)

Reply:
> It seems unlikely or impossible to report diagnostics on code you haven't analyzed

You definitely can do so. For example, if you have a CompilationEnd analyzer that wants to analyze only non-generated code, but report diagnostics anywhere in the compilation (including in generated code), then specifying only `ReportDiagnostics` would be required. I agree this is not a common analyzer configuration, but is theoretically possible.

We introduced `GeneratedCodeAnalysisFlags` as a `Flags` enum to provide complete flexibility to analyzer authors on whether or not to receive callbacks and/or report diagnostics for generated code.